test/e2e/internal/notation/registry.go

```diff
@@ -0,0 +1,169 @@
+package notation
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"math/rand"
+	"net/url"
+	"os"
+	"path/filepath"
+	"time"
+
+	"oras.land/oras-go/v2"
+	"oras.land/oras-go/v2/content/oci"
+	"oras.land/oras-go/v2/registry"
+	"oras.land/oras-go/v2/registry/remote"
+	"oras.land/oras-go/v2/registry/remote/auth"
+)
+
+type Registry struct {
+	Host     string
+	Username string
+	Password string
+}
+
+// CreateArtifact copies a local OCI layout to the a registry to create
+// a new artifact with a new repository.
+//
+// srcRepoName is the repo name in ./testdata/registry/oci_layout folder.
+// destRepoName is the repo name to be created in the registry.
+func (r *Registry) CreateArtifact(srcRepoName, destRepoName string) (*Artifact, error) {
+	ctx := context.Background()
+	// create a local store from OCI layout directory.
+	srcStore, err := oci.NewFromFS(ctx, os.DirFS(filepath.Join(OCILayoutPath, srcRepoName)))
+	if err != nil {
+		return nil, err
+	}
+
+	// create the artifact struct
+	artifact := &Artifact{
+		Registry: r,
+		Repo:     destRepoName,
+		Tag:      TestTag,
+	}
+	if err := artifact.Validate(); err != nil {
+		panic(err)
+	}
+
+	// create the remote.repository
+	destRepo, err := newRepository(artifact.ReferenceWithTag())
+	if err != nil {
+		return nil, err
+	}
+
+	// copy data
+	desc, err := oras.ExtendedCopy(ctx, srcStore, artifact.Tag, destRepo, "", oras.DefaultExtendedCopyOptions)
```

Comment:
Updated to use oras-go ExtendedCopy. Please have a look when you can. Thanks. @qweeah